### PR TITLE
feat(controller)!: Updates the resource duration calculation. Fixes #2934

### LINKED ISSUE #2934: Enhanced resource duration
* Import the updated algo from `usage` branch.
* Enable by default.
* Display denominator.

See #2219 and #1066

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to USERS.md.

Fixes #2934

# Changes

* This now defaults one and the config item name is changed. 
* The denominator for memory is now 100Mi rather than 1Gi so smaller memory usages are calculated.
* I fixed a bug in the calculation.
* The denominator is displayed in the UI.

# Screenshots

Screen Shot 2020-05-03 at 1 34 01 PM
Screen Shot 2020-05-03 at 1 35 48 PM

### CODE REVIEW

**Comment by alexec** [comment]
@elgalu, @gaigepr, @seichten, @lemor303442, @paguos, @goern, @wangadong, @phelinor, @ddseapy, @zhujl1991 can I please request your feedback on this change?

**Comment by alexec** [comment]
@elgalu, @gaigepr, @seichten, @lemor303442, @paguos, @goern, @wangadong, @phelinor, @zhujl1991 bump!

File: config/config.go [hunk 1]
```diff
@@ -56,9 +56,6 @@ type Config struct {
 	// MetricsConfig specifies configuration for metrics emission
 	MetricsConfig PrometheusConfig `json:"metricsConfig,omitempty"`
 
-	// FeatureFlags for general/experimental features
-	FeatureFlags FeatureFlags `json:"featureFlags,omitempty"`
```
Comment (simster7) [inline config/config.go:60]: Are we getting rid of feature flags entirely? Wouldn't we want to keep this for future features?

File: docs/resource-duration.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ information. This is intended to be an **indicative but not accurate** value.
 
 ## Configuration
 
-This is only set if [the feature flags is enabled in config](workflow-controller-configmap.yaml).
+This is turned on only if [the `resourcesDuration` flag is enabled in config](workflow-controller-configmap.yaml).
```
Comment (simster7) [inline docs/resource-duration.md:12]: I don't see the `resourcesDuration` flag anywhere in the config? Did I miss it?

Also, should it be `resourceDuration` (singular)?

File: pkg/apis/workflow/v1alpha1/workflow_types.go [hunk 3]
```diff
@@ -978,7 +979,7 @@ func (in ResourcesDuration) Add(o ResourcesDuration) ResourcesDuration {
 func (in ResourcesDuration) String() string {
 	var parts []string
 	for n, d := range in {
-		parts = append(parts, fmt.Sprintf("%v*%s", d, n))
+		parts = append(parts, fmt.Sprintf("%v*(%s %s)", d, ResourceQuantityDenominator(n).String(), n))
```
Comment (simster7) [inline pkg/apis/workflow/v1alpha1/workflow_types.go:982]: This looks way better in my opinion

File: config/config.go (hunk 1 above)
Comment (alexec) [reply config/config.go:60]: I don't think it is bad design. You can use normal config item or a envvar? Both are easier and more flexible.

File: docs/resource-duration.md (hunk 2 above)
Comment (alexec) [reply docs/resource-duration.md:12]: it is meant to be plural - but this config item is gone now - I'll correct

File: config/config.go (hunk 1 above)
Comment (sarabala1979) [reply config/config.go:60]: Are we enabling resource duration by default?

**Pull request merged**